### PR TITLE
Problems With HTML5

### PR DESCRIPTION
**Pull request by halhenke**
Using the FooBar example under the "Component - All Examples" link (where FooBar is passed to the ComponentRouter as a config object) triggers the whole page to reload whereas using it by following the link "Component - FooBar" (where the config prop is the FooBar function) works fine.

**EDIT** nevermind - I updated component-router and now its working fine  :disappointed: :gun:

### CODE REVIEW

**Comment by nkbt** [comment]
Huh? You mean it now all works well? I'll have a look later today

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (nkbt) [inline bower.json]: you can remove bower config from this repo.

File: src/Quickstart/Chart.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        <Url query={{chart: 'bar'}}>Bar</Url>
+        &nbsp;|&nbsp;
+        <Url query={{chart: 'pie'}}>Pie</Url>
+
+        <ComponentRouter namespace="chart" config={{
+          [getDefault()]: 'bar',
+          bar: Bar,
+          pie: Pie
+        }}>
+          <Content />
```
Comment (nkbt) [inline src/Quickstart/Chart.js]: you can remove `Content` completely in fresh version of ComponentRouter `Component` will be rendered if there is no `Container` element.

So just 
```
<ComponentRouter namespace="chart" config={{
  [getDefault()]: 'bar',
  bar: Bar,
  pie: Pie
}} />
```

File: src/Quickstart/Data.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        &nbsp;|&nbsp;
+        <Url query={{data: 'destinations'}}>Destinations</Url>
+
+
+        <ComponentRouter namespace="data" config={{
+          [getDefault()]: 'sources',
+          sources: Sources,
+          destinations: Destinations
+        }}>
+          <Content />
```
Comment (nkbt) [inline src/Quickstart/Data.js]: Same here. You can remove it

File: src/SideBar.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import Router from 'react-router';
```
Comment (nkbt) [inline src/SideBar.js]: `import {Link} from 'react-router';`

Since you don't use `Router` anywhere.

File: src/example/Example.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import routes from './App';
+import Router from 'react-router';
+
+
+// Router.run(routes, Router.HashLocation, (Root) => {
+Router.run(routes, Router.HistoryLocation, (Root) => {
```
Comment (nkbt) [inline src/example/Example.js]: `Root => React.render(<Root/>, document.querySelector('#app'));`

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import FooBar from './FooBar/FooBar';
+import Quickstart from './Quickstart/Quickstart';
+
+export {FooBar, Quickstart};
+export * from './SideBar';
```
Comment (nkbt) [inline src/index.js]: That is confusing. What is actually exported? (I know that is the syntax, but be a bit more explicit here)

**Comment by nkbt** [comment]
Wow you took a pretty complex example! That is really cool.

I was initially thinking about having just a dead-simple app with 2 pages routed by ReactRouter, one of those would have FooBar routed by ComponentRouter.

In this way it would be much easier for random people to read it.

But I like this complex example as well... Maybe keep this one as `advanced` and create another super-simple one in addition? What do you reckon?

**Comment by halhenke** [comment]
Yeah that sounds like a good idea. Will make the changes and have 2 different examples.

Yeah its weird - I saw you had made changes with the 2.2.1 version and so I reinstalled from npm and everything began working. It seemed to be something coming from the React event handling code itself and was happening with the same example code from the main `component-router` (after all the react-router code was stripped out). So I'm not sure what was happening - I'm just glad its over :-)

**Comment by nkbt** [comment]
I don't think anything is fixed now... I am going to look at that tomorrow. For some reason `pathname` is empty in the Store when click on the link happens. That is strange.

**Comment by halhenke** [comment]
Really? :disappointed: I'll do a fresh install straight from this repo and see what happens. 

One other thing I noted was that when it wasnt working for me the React Chrome plugin didn't show any Components loaded at all but now its working as expected.

**Comment by halhenke** [comment]
Was poking around and it seems that the empty pathname comes from the fact that pathname is part of Urls state

```javascript
  getInitialState() {
    return {
      query: Store.getQuery(),
      pathname: Store.getPathname()
    };
  },

  onChange() {
    this.setState({
      query: Store.getQuery(),
      pathname: Store.getPathname()
    });
  },
```
but when we dispatch a Navigate Action to the Store we pass the Url's props as the argument:

```javascript
  onClick(event) {
    event.preventDefault();
    Store.dispatch(ActionCreator.navigateTo(this.props));
  },
```
while the NavigateTo Action definitely expects both a pathname and a query as arguments (query seems to be in both props and state)? Figure you will probably want to handle this in [in-flux/component-router/issues/64]() ?

**Comment by nkbt** [comment]
I think it makes sense to remove pathname manipulations. I've already done almost all work there, but merge was delayed by redux refactor. Removing pathname should definitely make things easier to fix.

**Comment by halhenke** [comment]
Oh yeah I definitely agree. Or at least I'm pretty sure that the benefits of dealing purely with state in the key/val/query way outweigh the benefits of having it around. And its definitely in keeping with the philosophy of building something that isnt hierarchical like react-router. In that case using RR & CC together shouldnt involve much configuration/thought as long as people avoid using the query prop in their RR links....I hope :-)

**Comment by nkbt** [comment]
That's what I was thinking about too!



-- 
Sincerely yours, Nikita Butenko
Google Talk: nik@butenko.me
Skype: nikita.butenko
Mobile: +61 450 264 801

**Comment by halhenke** [comment]
Great minds... :wink:

**Pull request merged**